hibernate-core/src/main/java/org/hibernate/annotations/GeneratedColumn.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.annotations;
+
+import org.hibernate.tuple.GeneratedAlwaysValueGeneration;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Specifies that a column is defined using a DDL {@code generated always as} clause
+ * or equivalent, and whether Hibernate should fetch the generated value from the
+ * database.
+ *
+ * @author Gavin King
+ *
+ * @see ColumnDefault
+ */
+@Target( {FIELD, METHOD} )
+@Retention( RUNTIME )
+@ValueGenerationType(generatedBy = GeneratedAlwaysValueGeneration.class)
+public @interface GeneratedColumn {
```

Comment:
Wouldn't it be better to name it `ColumnGenerated` for consistency sake with `ColumnDefault`?

Reply:
I went down that path originally, but eventually settled on this, _after_ I realized that it's really not the same as `@ColumnDefault` at all. They're actually much more different than they seemed at first blush.